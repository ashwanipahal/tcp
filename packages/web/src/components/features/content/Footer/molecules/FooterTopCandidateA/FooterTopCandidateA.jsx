import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  RichText,
  Col,
  Row,
  BodyCopy,
  TextItems,
} from '@tcp/core/src/components/common/atoms';
import { reduxForm } from 'redux-form';
import { Grid } from '@tcp/core/src/components/common/molecules';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { formatPhoneNumber } from '@tcp/core/src/utils/formValidation/phoneNumber';
import { getLocator } from '@tcp/core/src/utils';
import ClickTracker from '@tcp/web/src/components/common/atoms/ClickTracker';

import SocialMediaLinks from '../SocialMediaLinks';
/* TODO move to component itself */
import style from '../../Footer.style';

import FooterTopSignUpForm from '../FooterTopSignUpForm';

const emailSignupFieldName = 'signup';
const emailSignupSecondBrand = 'isEmailOptInSecondBrand';
const FooterTopEmailSignUpForm = reduxForm({
  form: 'FooterTopEmailSignUpForm', // a unique identifier for this form
  initialValues: {
    [emailSignupFieldName]: '',
    [emailSignupSecondBrand]: false,
  },
})(FooterTopSignUpForm);

const smsSignupFieldName = 'footerTopSmsSignup';
const textSignupSecondBrand = 'isTextOptInSecondBrand';
const FooterTopSmsSignUpForm = reduxForm({
  form: 'FooterTopSmsSignUpForm', // a unique identifier for this form
  initialValues: {
    [smsSignupFieldName]: '',
    [textSignupSecondBrand]: false,
  },
})(FooterTopSignUpForm);

class FooterTopCandidateA extends React.PureComponent {
  render() {
    const {
      emailSignup,
      emailSignupLabels,
      smsSignup,
      smsSignupLabels,
      socialMediaLinks,
      referAFriendButtonLabels,
      referAFriend,
      validateSignupEmail,
      validateSignupSmsPhoneNumber,
      submitEmailSubscription,
      submitSmsSubscription,
      emailSubscription,
      smsSubscription,
      openEmailSignUpModal,
      openSmsSignUpModal,
      isNavigationFooter,
    } = this.props;

    return (
      <Grid className="footer_top_candidate_a content-wrapper">
        <Row>
          {/* ------------ Email Sign Up starts here ----------------- */}
          <Col
            className="col-md-half-width"
            colSize={{
              large: 4,
              medium: 4,
              small: 6,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <BodyCopy
              component="h2"
              fontWeight="black"
              fontSize="fs15"
              className="heading_text email-sign-up"
              dataLocator="email_promo_text"
            >
              <TextItems textItems={emailSignup.textItems} />
            </BodyCopy>
            <FooterTopEmailSignUpForm
              labels={emailSignupLabels}
              validateForm={validateSignupEmail}
              onFormSubmit={submitEmailSubscription}
              subscription={emailSubscription}
              openSuccessModal={openEmailSignUpModal}
              dataLocators={{
                submitButton: 'email_submit_btn',
                inputField: 'enter_email_text_field',
                errorDataLocator: 'email_error_message',
                checkBox_gym: 'check_box_gym_opt_in',
                checkBox_tcp: 'check_box_tcp_opt_in',
              }}
              fieldName={emailSignupFieldName}
              secondFieldName={emailSignupSecondBrand}
            />

            <BodyCopy fontFamily="secondary" textAlign="center" fontSize={['fs9', 'fs9', 'fs12']}>
              <RichText richTextHtml={emailSignupLabels.lbl_SignUp_termsTextLabel} />
            </BodyCopy>
            <div>
              <div className="divider hide-in-medium-up" />
            </div>
          </Col>

          {/* ---------- Email Signup ends here ---------- */}

          {/* ---------- SMS Sign Up starts here ---------- */}
          <Col
            className="col-md-half-width"
            colSize={{
              large: 4,
              medium: 4,
              small: 6,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <BodyCopy
              component="h2"
              fontWeight="black"
              fontSize="fs15"
              className="heading_text sms_sign_up"
              dataLocator="sms_promo_text"
            >
              <TextItems textItems={smsSignup.textItems} />
            </BodyCopy>
            <FooterTopSmsSignUpForm
              labels={smsSignupLabels}
              fieldName={smsSignupFieldName}
              fieldProps={{
                normalize: formatPhoneNumber,
              }}
              validateForm={validateSignupSmsPhoneNumber}
              onFormSubmit={submitSmsSubscription}
              subscription={smsSubscription}
              openSuccessModal={openSmsSignUpModal}
              dataLocators={{
                submitButton: 'sms_submit_btn',
                inputField: 'sms_field',
                errorDataLocator: 'sms_error_message',
                checkBox_gym: 'sms_gym_opt_in',
                checkBox_tcp: 'sms_tcp_opt_in',
              }}
              secondFieldName={textSignupSecondBrand}
            />

            <BodyCopy fontFamily="secondary" textAlign="center" fontSize={['fs9', 'fs9', 'fs12']}>
              <RichText richTextHtml={smsSignupLabels.lbl_SignUp_termsTextLabel} />
            </BodyCopy>
          </Col>
          <div className="divider hide-in-large-up" />
          {/* ---------- SMS Signup ends here ------------ */}

          {/* ---------- Refer a friend start here-------- */}

          <Col
            className="refer_a_friend_desktop col-md-half-width"
            colSize={{
              large: 4,
              medium: 4,
              small: 6,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <Grid>
              <Row fullBleed className="flex-align-center">
                <Col
                  className="col-md-half-width"
                  colSize={{
                    large: 6,
                    medium: 4,
                    small: 7,
                  }}
                  ignoreGutter={{
                    small: true,
                  }}
                >
                  <BodyCopy
                    component="h2"
                    fontWeight="black"
                    fontSize="fs15"
                    className="heading_text refer-a-friend"
                  >
                    <TextItems textItems={referAFriend.textItems} />
                  </BodyCopy>
                </Col>
                <Col
                  colSize={{
                    large: 6,
                    medium: 4,
                    small: 6,
                  }}
                  ignoreGutter={{
                    small: false,
                  }}
                  className="candidate_a_inline_container_button col-md-half-width"
                >
                  <ClickTracker
                    as={Button}
                    id={
                      isNavigationFooter
                        ? 'extole_zone_global_navigation_footer'
                        : 'extole_zone_global_footer'
                    }
                    buttonVariation="variable-width"
                    data-locator={getLocator('refer_friend')}
                    className="refer_a_friend_button"
                    clickData={{
                      customEvents: ['event22', 'event80'],
                      pageShortName: 'content:referafriend  confirmation',
                      pageName: 'content:email confirmation',
                    }}
                  >
                    {referAFriendButtonLabels.text}
                  </ClickTracker>
                </Col>
              </Row>
              <div className="divider hide-in-medium-down" />
              <Row fullBleed className="hide-in-medium-down">
                <Col
                  className=""
                  colSize={{
                    large: 12,
                    medium: 12,
                    small: 12,
                  }}
                  ignoreGutter={{
                    small: true,
                  }}
                >
                  <div className="divider hide-in-medium-up" />
                  <SocialMediaLinks
                    {...socialMediaLinks}
                    className="footer_top_candidate_a_social_links"
                  />
                </Col>
              </Row>
            </Grid>
          </Col>
          {/* ---------- Refer a friend ends here-------- */}
          <div className="divider hide-in-medium-up" />
          <Col
            className="hide-in-large-up refer_a_frient_last_colm col-md-half-width"
            colSize={{
              large: 0,
              medium: 4,
              small: 12,
            }}
            ignoreGutter={{
              small: true,
            }}
          >
            <SocialMediaLinks {...socialMediaLinks} className="social-media-links" />
          </Col>
        </Row>
      </Grid>
    );
  }
}

FooterTopCandidateA.propTypes = {
  className: PropTypes.string.isRequired,
  emailSignup: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  smsSignup: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  emailSignupLabels: PropTypes.shape({
    lbl_SignUp_placeholderText: PropTypes.string,
    lbl_SignUp_validationErrorLabel: PropTypes.string,
    lbl_SignUp_termsTextLabel: PropTypes.string,
    lbl_SignUp_submitButtonLabel: PropTypes.string,
    lbl_SignUp_gymSignUpLabel: PropTypes.string,
    lbl_SignUp_tcpSignUpLabel: PropTypes.string,
  }),
  smsSignupLabels: PropTypes.shape({
    lbl_SignUp_placeholderText: PropTypes.string,
    lbl_SignUp_validationErrorLabel: PropTypes.string,
    lbl_SignUp_termsTextLabel: PropTypes.string,
    lbl_SignUp_submitButtonLabel: PropTypes.string,
  }),
  socialMediaLinks: PropTypes.arrayOf(
    PropTypes.shape({
      icon_class: PropTypes.string,
      title: PropTypes.string,
    })
  ).isRequired,
  referAFriend: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  referAFriendButtonLabels: PropTypes.shape({
    title: PropTypes.string,
    text: PropTypes.string,
  }),
  showError: PropTypes.bool,
  isEmailValid: PropTypes.bool,
  validateSignupEmail: PropTypes.func,
  validateSignupSmsPhoneNumber: PropTypes.func,
  submitEmailSubscription: PropTypes.func,
  submitSmsSubscription: PropTypes.func,
  openEmailSignUpModal: PropTypes.func,
  openSmsSignUpModal: PropTypes.func,
  emailSubscription: PropTypes.shape({}),
  smsSubscription: PropTypes.shape({}),
  isNavigationFooter: PropTypes.bool,
};

FooterTopCandidateA.defaultProps = {
  emailSignup: {
    text: '',
    title: '',
  },
  smsSignup: {
    text: '',
    title: '',
  },
  referAFriendButtonLabels: {
    title: '',
    text: '',
  },
  emailSignupLabels: {
    lbl_SignUp_placeholderText: '',
    lbl_SignUp_validationErrorLabel: '',
    lbl_SignUp_termsTextLabel: '',
    lbl_SignUp_submitButtonLabel: '',
    lbl_SignUp_gymSignUpLabel: '',
    lbl_SignUp_tcpSignUpLabel: '',
  },
  smsSignupLabels: {
    lbl_SignUp_placeholderText: '',
    lbl_SignUp_validationErrorLabel: '',
    lbl_SignUp_termsTextLabel: '',
    lbl_SignUp_submitButtonLabel: '',
  },
  referAFriend: {
    title: '',
    text: '',
  },
  showError: false,
  isEmailValid: false,
  validateSignupEmail: () => Promise.resolve(),
  validateSignupSmsPhoneNumber: () => Promise.resolve(),
  submitEmailSubscription: () => {},
  submitSmsSubscription: () => {},
  openEmailSignUpModal: () => {},
  openSmsSignUpModal: () => {},
  emailSubscription: {},
  smsSubscription: {},
  isNavigationFooter: false,
};

export default withStyles(FooterTopCandidateA, style);

export { FooterTopCandidateA as FooterTopCandidateAVanilla };
