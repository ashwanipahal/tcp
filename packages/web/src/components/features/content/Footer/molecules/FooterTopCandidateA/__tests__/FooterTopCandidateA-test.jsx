import React from 'react';
import { shallow } from 'enzyme';
import { FooterTopCandidateAVanilla } from '../FooterTopCandidateA';

describe('FooterTopCandidateAVanilla component', () => {
  it('renders correctly', () => {
    const component = shallow(<FooterTopCandidateAVanilla />);
    expect(component).toMatchSnapshot();
  });
});
