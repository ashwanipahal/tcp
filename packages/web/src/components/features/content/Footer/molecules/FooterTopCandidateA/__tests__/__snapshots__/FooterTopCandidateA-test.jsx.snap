// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterTopCandidateAVanilla component renders correctly 1`] = `
<Styled(Grid)
  className="footer_top_candidate_a content-wrapper"
>
  <Styled(Row)>
    <Styled(Col)
      className="col-md-half-width"
      colSize={
        Object {
          "large": 4,
          "medium": 4,
          "small": 6,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(BodyCopy)
        className="heading_text email-sign-up"
        color="text.primary"
        component="h2"
        dataLocator="email_promo_text"
        fontFamily="primary"
        fontSize="fs15"
        fontWeight="black"
        letterSpacing="normal"
        lineHeight="normal"
        link={null}
        textAlign="left"
      >
        <TextItems />
      </Styled(BodyCopy)>
      <ReduxForm
        dataLocators={
          Object {
            "checkBox_gym": "check_box_gym_opt_in",
            "checkBox_tcp": "check_box_tcp_opt_in",
            "errorDataLocator": "email_error_message",
            "inputField": "enter_email_text_field",
            "submitButton": "email_submit_btn",
          }
        }
        destroyOnUnmount={true}
        enableReinitialize={false}
        fieldName="signup"
        forceUnregisterOnUnmount={false}
        form="FooterTopEmailSignUpForm"
        getFormState={[Function]}
        initialValues={
          Object {
            "isEmailOptInSecondBrand": false,
            "signup": "",
          }
        }
        keepDirtyOnReinitialize={false}
        labels={
          Object {
            "lbl_SignUp_gymSignUpLabel": "",
            "lbl_SignUp_placeholderText": "",
            "lbl_SignUp_submitButtonLabel": "",
            "lbl_SignUp_tcpSignUpLabel": "",
            "lbl_SignUp_termsTextLabel": "",
            "lbl_SignUp_validationErrorLabel": "",
          }
        }
        onFormSubmit={[Function]}
        openSuccessModal={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        secondFieldName="isEmailOptInSecondBrand"
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        subscription={Object {}}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        validateForm={[Function]}
      />
      <Styled(BodyCopy)
        className={null}
        color="text.primary"
        component="p"
        fontFamily="secondary"
        fontSize={
          Array [
            "fs9",
            "fs9",
            "fs12",
          ]
        }
        fontWeight="regular"
        letterSpacing="normal"
        lineHeight="normal"
        link={null}
        textAlign="center"
      >
        <Styled(RichText)
          richTextHtml=""
        />
      </Styled(BodyCopy)>
      <div>
        <div
          className="divider hide-in-medium-up"
        />
      </div>
    </Styled(Col)>
    <Styled(Col)
      className="col-md-half-width"
      colSize={
        Object {
          "large": 4,
          "medium": 4,
          "small": 6,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(BodyCopy)
        className="heading_text sms_sign_up"
        color="text.primary"
        component="h2"
        dataLocator="sms_promo_text"
        fontFamily="primary"
        fontSize="fs15"
        fontWeight="black"
        letterSpacing="normal"
        lineHeight="normal"
        link={null}
        textAlign="left"
      >
        <TextItems />
      </Styled(BodyCopy)>
      <ReduxForm
        dataLocators={
          Object {
            "checkBox_gym": "sms_gym_opt_in",
            "checkBox_tcp": "sms_tcp_opt_in",
            "errorDataLocator": "sms_error_message",
            "inputField": "sms_field",
            "submitButton": "sms_submit_btn",
          }
        }
        destroyOnUnmount={true}
        enableReinitialize={false}
        fieldName="footerTopSmsSignup"
        fieldProps={
          Object {
            "normalize": [Function],
          }
        }
        forceUnregisterOnUnmount={false}
        form="FooterTopSmsSignUpForm"
        getFormState={[Function]}
        initialValues={
          Object {
            "footerTopSmsSignup": "",
            "isTextOptInSecondBrand": false,
          }
        }
        keepDirtyOnReinitialize={false}
        labels={
          Object {
            "lbl_SignUp_placeholderText": "",
            "lbl_SignUp_submitButtonLabel": "",
            "lbl_SignUp_termsTextLabel": "",
            "lbl_SignUp_validationErrorLabel": "",
          }
        }
        onFormSubmit={[Function]}
        openSuccessModal={[Function]}
        persistentSubmitErrors={false}
        pure={true}
        secondFieldName="isTextOptInSecondBrand"
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        submitAsSideEffect={false}
        subscription={Object {}}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        validateForm={[Function]}
      />
      <Styled(BodyCopy)
        className={null}
        color="text.primary"
        component="p"
        fontFamily="secondary"
        fontSize={
          Array [
            "fs9",
            "fs9",
            "fs12",
          ]
        }
        fontWeight="regular"
        letterSpacing="normal"
        lineHeight="normal"
        link={null}
        textAlign="center"
      >
        <Styled(RichText)
          richTextHtml=""
        />
      </Styled(BodyCopy)>
    </Styled(Col)>
    <div
      className="divider hide-in-large-up"
    />
    <Styled(Col)
      className="refer_a_friend_desktop col-md-half-width"
      colSize={
        Object {
          "large": 4,
          "medium": 4,
          "small": 6,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(Grid)>
        <Styled(Row)
          className="flex-align-center"
          fullBleed={true}
        >
          <Styled(Col)
            className="col-md-half-width"
            colSize={
              Object {
                "large": 6,
                "medium": 4,
                "small": 7,
              }
            }
            ignoreGutter={
              Object {
                "small": true,
              }
            }
          >
            <Styled(BodyCopy)
              className="heading_text refer-a-friend"
              color="text.primary"
              component="h2"
              fontFamily="primary"
              fontSize="fs15"
              fontWeight="black"
              letterSpacing="normal"
              lineHeight="normal"
              link={null}
              textAlign="left"
            >
              <TextItems />
            </Styled(BodyCopy)>
          </Styled(Col)>
          <Styled(Col)
            className="candidate_a_inline_container_button col-md-half-width"
            colSize={
              Object {
                "large": 6,
                "medium": 4,
                "small": 6,
              }
            }
            ignoreGutter={
              Object {
                "small": false,
              }
            }
          >
            <Connect(Component)
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "rules": Array [
                      "
  ",
                      "
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  cursor: ",
                      [Function],
                      ";
  opacity: ",
                      [Function],
                      ";
  text-transform: uppercase;
  min-height: 42px;
  letter-spacing: 0.93px;
  font-size: ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      "
  ",
                      [Function],
                      "
  ",
                      [Function],
                      ";
  ",
                      [Function],
                      ";

  ",
                      [Function],
                      ";

  ",
                      [Function],
                      ";

  &:focus {
    background: ",
                      [Function],
                      ";
  }
  &:hover:not[disabled] {
    background: ",
                      [Function],
                      ";
  }

  ",
                      [Function],
                      ";

  ",
                      [Function],
                      ";

  ",
                      [Function],
                      ";

  ",
                      [Function],
                      "

  @media ",
                      [Function],
                      " {
    ",
                      [Function],
                      ";
    ",
                      [Function],
                      ";
  }
  ",
                      [Function],
                      "

  ",
                      [Function],
                      "
      ",
                      [Function],
                      ";
      ",
                      [Function],
                      ";

",
                      ";
",
                    ],
                  },
                  "displayName": "Styled(Button)",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": [Function],
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              buttonVariation="variable-width"
              className="refer_a_friend_button"
              clickData={
                Object {
                  "customEvents": Array [
                    "event22",
                    "event80",
                  ],
                  "pageName": "content:email confirmation",
                  "pageShortName": "content:referafriend  confirmation",
                }
              }
              data-locator="global_referfriend"
              id="extole_zone_global_footer"
              pageData={
                Object {
                  "pageShortName": "content:referafriend  confirmation",
                }
              }
            />
          </Styled(Col)>
        </Styled(Row)>
        <div
          className="divider hide-in-medium-down"
        />
        <Styled(Row)
          className="hide-in-medium-down"
          fullBleed={true}
        >
          <Styled(Col)
            className=""
            colSize={
              Object {
                "large": 12,
                "medium": 12,
                "small": 12,
              }
            }
            ignoreGutter={
              Object {
                "small": true,
              }
            }
          >
            <div
              className="divider hide-in-medium-up"
            />
            <Styled(SocialMediaLinks)
              className="footer_top_candidate_a_social_links"
            />
          </Styled(Col)>
        </Styled(Row)>
      </Styled(Grid)>
    </Styled(Col)>
    <div
      className="divider hide-in-medium-up"
    />
    <Styled(Col)
      className="hide-in-large-up refer_a_frient_last_colm col-md-half-width"
      colSize={
        Object {
          "large": 0,
          "medium": 4,
          "small": 12,
        }
      }
      ignoreGutter={
        Object {
          "small": true,
        }
      }
    >
      <Styled(SocialMediaLinks)
        className="social-media-links"
      />
    </Styled(Col)>
  </Styled(Row)>
</Styled(Grid)>
`;
